build.gradle.kts

```diff
@@ -172,7 +172,21 @@ tasks {
         //TODO inject changelog into plugin.xml change-notes
     }
 
+    // TODO: Once LSP4IJ no longer supports 2023.*, the hotswap JVM args can be added unconditionally
+    fun supportsEnhancedClassRedefinition(): Boolean {
+        val platformVersion = properties("platformVersion").getOrNull()
+        return when (platformVersion) {
+            // Versions before 2024.1 don't support enhanced class redefinition
+            "2023.2", "2023.3" -> false
```

Comment:
I've explicitly verified the correct behavior in 2023.2, 2023.3, 2024.1, 2024.2, and 2024.3. By "correct behavior", I mean that `runIde` actually does run an IDE instance of the specified `platformVersion`, and once started, only simple classloader hotswaps are supported on 2023.* and enhanced classloader hotswaps are supported on 2024.*.

Reply:
that's great news, thanks for sharing!